### PR TITLE
[fuzz] Reorganize fuzz/ to be mechanically regenerate-able

### PR DESCRIPTION
**Pull request by mcy**
Writing fuzz tests is a bit of a pain since it's a bit easy to trip over the generated ones. This replaces that issue with automation. This change is otherwise a no-op.

Woooo automation!

Signed-off-by: Miguel Young de la Sota <mcyoung@google.com>

### CODE REVIEW

**Comment by mcy** [comment]
Dear reviewer: the main file of interest is `generate_fuzz_tests.py`. Everything else is either machine generated or virtually unchanged.

File: fuzz/generate_proto_fuzz.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+  cargo_toml = reset_generated_config(FUZZ_CARGO, CARGO_END)
+  cargo_tpl = CARGO_TPL.read_text()
+
+  fuzz_yml = reset_generated_config(FUZZ_YML, YML_END)
+  yml_tpl = YML_TPL.read_text()
+
+  count = 0
+  for ty in FUZZ_CONFIG.read_text().split('\n'):
+    if not ty or ty.startswith('#'): continue
```
Comment (cfrantz) [inline fuzz/generate_proto_fuzz.py:71]: nit: continue on next line? (here and below)

File: fuzz/generate_proto_fuzz.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  eprint(f'Cargo exited with {out.returncode}')
+  if out.returncode != 0:
+    eprint(f'Fuzz build failed!')
+    sys.exit(out.returncode)
+
+def reset_generated_config(path, delim):
+  text = path.read_text()
+  return text[:text.index(delim)] + delim + '\n'
+
+def main():
```
Comment (cfrantz) [inline fuzz/generate_proto_fuzz.py:56]: Just a style thing:  Out of loyalty to C, I normally write main as:

```
def main(argv):
   ...

if __name__ == '__main__':
    main(sys.argv)
```

**Review by cfrantz — APPROVED** [review]

File: fuzz/generate_proto_fuzz.py (hunk 1 above)
Comment (mcy) [reply fuzz/generate_proto_fuzz.py:71]: I liked the compactness but sure. =P

File: fuzz/generate_proto_fuzz.py (hunk 2 above)
Comment (mcy) [reply fuzz/generate_proto_fuzz.py:56]: Hehe, sure.

**Pull request merged**